qc_tests/streaks.py

```diff
@@ -93,102 +170,354 @@ def get_repeating_string_threshold(obs_var, config_dict, plots=False, diagnostic
+                                                        diagnostics=diagnostics)
+
+        long_enough_streaks, = np.where(year_repeated_streak_lengths >= MIN_STREAK_LENGTH_FOR_EXCESS_FREQUENCY)
+
+        proportion = np.sum(year_repeated_streak_lengths[long_enough_streaks])/len(this_var.data[locs].compressed())
+
+        if proportion <= threshold:
+            # Move on to next year
+            continue
+
+        # all identified streaks (>= MIN_STREAK_LENGTH_FOR_EXCESS_FREQUENCY in a row)
+        for streak in long_enough_streaks:
+            start = int(np.sum(grouped_diffs[:streaks[streak], 1]))
+            end = start + int(grouped_diffs[streaks[streak], 1]) + 1
+
+            year_flags[unmasked[start : end]] = "x"
+
+            if plots:
+                plot_streak(masked_times, this_var.data[locs], obs_var.units, start, end)
+
+        flags[locs] = year_flags
+
+    # Write into original object (the one with calm periods)
+    obs_var.flags = utils.insert_flags(obs_var.flags, flags)
+
+    logger.info(f"Excess Repeated streaks {this_var.name}")
+    logger.info(f"   Cumulative number of flags set: {len(np.ma.nonzero(flags != '')[0])}")
 
     return # excess_repeating_value
 
 
 #************************************************************************
-def day_repeat():
+def repeating_day(obs_var:utils.Meteorological_Variable, station: utils.Station,
+                  config_dict: dict, determine_threshold=False,
+                  plots: bool = False, diagnostics: bool = False) -> None:
+
+    # any complete repeat of 24hs (as long as not "straight streak")
 
-    # any complete repeat of 24hs (as long as not "straight string")
+    set_flags = not determine_threshold    
 
+    if set_flags:
+        flags = np.array(["" for _ in obs_var.data])
+        # retrieve the threshold and store
+        try:
+            threshold = config_dict[f"STREAK-{obs_var.name}"]["DayRepeat"]
+        except KeyError:
+            # no threshold set
+            # For this routine, can't set to call the setting routine as it is itself
+            return
+
+    # sort into array of days
+    # check each day and compare to previous. HadISD used fixed threshold, but could do one pass to get, second to apply/?
+    all_lengths = []
+    streak_length = 0
+    previous_day_data = np.array([])
+    streak_locs = np.array([], dtype=(int))
+
+    for year in np.unique(station.years):
+        for month in np.unique(station.months):
+            for day in np.unique(station.days):
+                try:
+                    # if Datetime doesn't throw an error, then valid date
+                    _ = dt.datetime(year, month, day)
+                except ValueError:
+                    # not a valid day (e.g. Leap years, short months etc)
+                    continue
+
+                this_day_locs, = np.where(np.logical_and.reduce((station.years == year,
+                                                        station.months == month,
```

Comment:
Clever! Check use of np.where if you move to numpy 2.0?

Reply:
Done - np.nonzero(x in array) is the exact equivalent of np.where(x in array).  np.where() does other things if you e.g. pass in extra arguments, which I've rarely used, and will ensure that I replace each instance manually to check.